migration/identity-migration/src/main/java/io/camunda/migration/identity/midentity/ConsoleClient.java

```diff
@@ -0,0 +1,109 @@
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.migration.identity.midentity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import io.camunda.migration.identity.config.IdentityMigrationProperties;
+import java.util.List;
+import org.springframework.web.client.RestTemplate;
+
+public class ConsoleClient {
+
+  private static final String MIGRATION_DATA_ENDPOINT =
+      "/external/organizations/{0}/clusters/{1}/migrationData/{2}";
+
+  private final IdentityMigrationProperties properties;
+  private final RestTemplate restTemplate;
+
+  public ConsoleClient(
+      final IdentityMigrationProperties properties, final RestTemplate restTemplate) {
+    this.properties = properties;
+    this.restTemplate = restTemplate;
+  }
+
+  public Members fetchMembers() {
+    return restTemplate.getForObject(
+        MIGRATION_DATA_ENDPOINT,
+        Members.class,
+        properties.getOrganizationId(),
+        properties.getConsole().getClusterId(),
+        properties.getConsole().getInternalClientId());
+  }
+
+  public record Members(List<Member> members, List<Client> clients) {}
+
+  public record Client(String name, String clientId, List<Permission> permissions) {}
+
+  public record Member(String originalUserId, List<Role> roles, String email, String name) {}
+
+  public enum Permission {
+    ZEEBE("Zeebe"),
+    OPERATE("Operate"),
+    TASKLIST("Tasklist"),
+    OPTIMIZE("Optimize"),
+    SECRETS("Secrets");
+
+    private final String name;
+
+    Permission(final String name) {
+      this.name = name;
+    }
+
+    @JsonValue
+    public String getName() {
+      return name;
+    }
+
+    @JsonCreator
+    public static Permission fromValue(String value) {
+      for (Permission permission : Permission.values()) {
+        if (permission.name.equalsIgnoreCase(value)) {
+          return permission;
+        }
+      }
+      throw new IllegalArgumentException("Unknown permission: " + value);
+    }
+  }
+
+  public enum Role {
+    MEMBER("member"),
+    ADMIN("admin"),
+    ORGANIZATION_ADMIN("organizationadmin"),
+    ORGANIZATION_OWNER("organizationowner"),
+    OWNER("owner"),
+    SUPPORT_AGENT("supportagent"),
+    OPERATIONS_ENGINEER("operationsengineer"),
+    TASK_USER("taskuser"),
+    ANALYST("analyst"),
+    DEVELOPER("developer"),
+    VISITOR("visitor"),
+    MODELER("modeler");
+
+    private final String name;
+
+    Role(final String name) {
+      this.name = name;
+    }
+
+    @JsonValue
+    public String getName() {
+      return name;
+    }
+
+    @JsonCreator
+    public static Role fromValue(String value) {
+      for (Role role : Role.values()) {
+        if (role.name.equalsIgnoreCase(value)) {
+          return role;
+        }
+      }
+      throw new IllegalArgumentException("Unknown role: " + value);
```

Comment:
🔧   I would be in favor of being more lean on unknown roles and not let parsing fail completely🤔  

As effectively it could change at any point in time out of our control and CI :D, it might be worth logging that we encountered an unhandled role but assuming it's not relevant for the scope of this migration, we should not fail on it. Would you agree?

Same for permissions, we are only interested in a subset there too.

Not sure what the best solution would be, we could delegate that to the service layer and make those strings here or add a `IGNORED` value to map those to + log + skip those in service layer.

In overall I'm in favor towards handling this explicitly in the service layer though, I wouldn't expect a client to change the response with the migration logic in mind.